contracts/oracles/default/CurveV2PriceOracle.sol

```diff
@@ -0,0 +1,143 @@
+    public
+    view
+    returns (
+      ICurvePool,
+      int128,
+      int128
+    )
+  {
+    for (uint256 i = 0; i < tokens.length; i++) {
+      ICurvePool pool = ICurvePool(poolFor[tokens[i]]);
+      int128 inputIndex = -1;
+      int128 outputIndex = -1;
+      int128 j = 0;
+      while (true) {
+        try pool.coins(uint256(uint128(j))) returns (address coin) {
+          if (coin == inputToken) inputIndex = j;
+          else if (coin == outputToken) outputIndex = j;
+          j++;
+        } catch {
+          break;
+        }
+
+        if (outputIndex > -1 && inputIndex > -1) {
+          return (pool, inputIndex, outputIndex);
+        }
+      }
+    }
+
+    return (ICurvePool(address(0)), int128(0), int128(0));
+  }
+
+  /**
+   * @notice Get the LP token price price for an underlying token address.
+   * @param underlying The underlying token address for which to get the price (set to zero address for ETH).
+   * @return Price denominated in ETH (scaled by 1e18).
+   */
+  function price(address underlying) external view override returns (uint256) {
+    return _price(underlying);
+  }
+
+  /**
+   * @notice Returns the price in ETH of the token underlying `cToken`.
+   * @dev Implements the `PriceOracle` interface for Fuse pools (and Compound v2).
+   * @return Price in ETH of the token underlying `cToken`, scaled by `10 ** (36 - underlyingDecimals)`.
+   */
+  function getUnderlyingPrice(ICErc20 cToken) external view override returns (uint256) {
+    address underlying = cToken.underlying();
+    // Comptroller needs prices to be scaled by 1e(36 - decimals)
+    // Since `_price` returns prices scaled by 18 decimals, we must scale them by 1e(36 - 18 - decimals)
+    return (_price(underlying) * 1e18) / (10**uint256(EIP20Interface(underlying).decimals()));
+  }
+
+  /**
+   * @dev Fetches the fair LP token price from Curve, with 18 decimals of precision.
+   * @param token The LP token contract address for price retrieval.
+   */
+  function _price(address token) internal view returns (uint256) {
+    address pool = poolFor[token];
+    require(address(pool) != address(0), "Token is not registered.");
+
+    address baseToken;
+    // Returns always coin(1) / coin(0)  [ e.g. USDC (1) / eUSDC (1) ]
+    uint256 exchangeRate = ICurveV2Pool(pool).price_oracle();
+
+    if (ICurvePool(pool).coins(0) == token) {
+      baseToken = ICurvePool(pool).coins(1);
+      // USDC / ETH
+      uint256 baseTokenPrice = BasePriceOracle(msg.sender).price(baseToken);
+      // USDC / ETH * eUSDC / USDC = eUSDC / ETH
+      return (baseTokenPrice * 10**18) / exchangeRate;
+    } else {
+      // if coin(1) is eUSDC, exchangeRate is USDC / eUSDC
+      baseToken = ICurvePool(pool).coins(0);
+      // USDC / ETH
+      uint256 baseTokenPrice = BasePriceOracle(msg.sender).price(baseToken);
+      // (USDC / ETH) *  (1 / (USDC / eUSDC)) = eUSDC / ETH
+      return (baseTokenPrice * exchangeRate) / 10**18;
```

Comment:
let's edit these comments, so that they make sense for the general case? why is assumed that the `baseToken` is at the `1` or the `0` index? Why is it assumed that the pool has only 2 underlying tokens?